pkgs/by-name/re/rerun/package.nix

```
@@ -49,7 +55,38 @@ rustPlatform.buildRustPackage rec {
   cargoBuildFlags = [ "--package rerun-cli" ];
   cargoTestFlags = [ "--package rerun-cli" ];
   buildNoDefaultFeatures = true;
-  buildFeatures = [ "native_viewer" ];
+  buildFeatures = [
+    "native_viewer"
+    "web_viewer"
+    "nasm"
+  ];
+
+  # Forward as a bash environment variable to the preBuild hook
+  inherit buildWebViewerFeatures;
+
+  # When web_viewer is compiled, the wasm webviewer first needs to be built
+  # If this doesn't exist, the build will fail. More information: https://github.com/rerun-io/rerun/issues/6028
+  # The command is taken from https://github.com/rerun-io/rerun/blob/dd025f1384f9944d785d0fb75ca4ca1cd1792f17/pixi.toml#L198C72-L198C187
+  # Note that cargoBuildFeatures reference what buildFeatures is set to in stdenv.mkDerivation,
+  # so that user can easily create an overlay to set cargoBuildFeatures to what he needs
+  preBuild = ''
+    if [[ " $cargoBuildFeatures " == *" web_viewer "* ]]; then
+      # transform the environment variable that is a space separated list into a comma separated list
+      buildWebViewerFeatures=$(echo $buildWebViewerFeatures | tr ' ' ',')
+      # Create the features option only if there are features to pass
+      buildWebViewerFeaturesCargoOption=""
+      if [[ ! -z "$buildWebViewerFeatures" ]]; then
+        buildWebViewerFeaturesCargoOption="--features $buildWebViewerFeatures"
+        echo "Features passed to the web viewer build: $buildWebViewerFeatures"
+      else
+        echo "No features will be passed to the web viewer build"
+      fi
+      echo "Building the wasm web viewer for rerun's web_viewer feature"
+      cargo run -p re_dev_tools -- build-web-viewer --no-default-features $buildWebViewerFeaturesCargoOption --release -g
```

Can quote the inputs; shouldn't cause trouble with the `--features=` form
```suggestion
      cargo run -p re_dev_tools -- build-web-viewer --no-default-features "$buildWebViewerFeaturesCargoOption" --release -g
```